src/plugins/intel_npu/src/al/include/intel_npu/al/icompiled_model.hpp

```diff
@@ -17,15 +18,17 @@ class ICompiledModel : public ov::ICompiledModel {
 public:
     using ov::ICompiledModel::ICompiledModel;
 
-    virtual const std::shared_ptr<const NetworkDescription>& get_network_description() const = 0;
+    virtual const NetworkDescription& get_network_description() const = 0;
```

Comment:
ND holds blob data. While the class is designed to prevent accidental copies, there might be reasons to hold the ND reference in some objects. shared_ptr allows that without a need for copy. Given how heavy that object is using shared_ptr is cheap in comparison and even if it might not be necessary to hold references to it I would rather keep shared_ptr in the interface to allow flexibility of the implementation.

I prefer to keep shared_ptr here to prevent accidenat